compose/progress_stream.py

```diff
@@ -1,20 +1,24 @@
 import json
-import os
 import codecs
 
+import six
+
 
 class StreamOutputError(Exception):
     pass
 
 
 def stream_output(output, stream):
-    is_terminal = hasattr(stream, 'fileno') and os.isatty(stream.fileno())
-    stream = codecs.getwriter('utf-8')(stream)
+    is_terminal = hasattr(stream, 'isatty') and stream.isatty()
+    if not six.PY3:
+        stream = codecs.getwriter('utf-8')(stream)
     all_events = []
     lines = {}
     diff = 0
 
     for chunk in output:
+        if six.PY3 and not isinstance(chunk, str):
+            chunk = chunk.decode('utf-8')
```

Comment:
It turns out docker-py has this same switch https://github.com/docker/docker-py/blob/master/docker/clientbase.py#L178, so I think this is correct.